### ISSUE TITLE
Error when building with nix

### ISSUE THREAD

**Issue by amitaibu** [digitallyinduced/ihp#1215]
Trying to follow https://ihp.digitallyinduced.com/Guide/deployment.html#building-with-nix

```
make prepare-optimized-nix-build

sed -i '' 's/haskellDeps/optimized = true;\n        haskellDeps/g' default.nix
sed: can't read s/haskellDeps/optimized = true;\n        haskellDeps/g: No such file or directory
make: *** [IHP/lib/IHP/Makefile.dist:224: prepare-optimized-nix-build] Error 2
```

I also tried `nix-build`. That seems to build, bu then there's no `RunProdServer`.

Even If I manually navigate to the nix store directory

```
[30 of 30] Compiling RunJobs          ( build/RunJobs.hs, build/RunUnoptimizedProdServer/RunJobs.o, build/RunUnoptimizedProdServer/RunJobs.dyn_o )
Linking build/bin/RunJobs ...
installing
/nix/store/n3zy2s63rz4jddqsrimyd3b1mf1yqcza-app
```

```
cd /nix/store/n3zy2s63rz4jddqsrimyd3b1mf1yqcza-app/bin
./RunProdServer

IHP_SESSION_SECRET=XXXXX
dirname: missing operand
Try 'dirname --help' for more information.
RunProdServerWithoutOptions: readCreateProcess: dirname $(which RunDevServer) (exit 1): failed
```

**Comment by mpscholten** [comment]
I guess the first issue is related to the `-i` flag of `sed`. Likely it's not supported on all platforms. But you can workaround this by adding `optimized = true;` manually before `haskellDeps = p: with p; [` in your `default.nix`.


To start the server binary the PWD has to be in `/nix/store/n3zy2s63rz4jddqsrimyd3b1mf1yqcza-app/lib`. So try this:

```bash
cd /nix/store/n3zy2s63rz4jddqsrimyd3b1mf1yqcza-app/lib
/nix/store/n3zy2s63rz4jddqsrimyd3b1mf1yqcza-app/bin/RunProdServer
```

**Comment by amitaibu** [comment]
It's working. I'd like to update the docs:

1. Is there a proper way of getting the nix store directory after the command, or just some text processing by bash?
2. What is the reason we need to be on `../lib`?

Also, what do you want to do about the failing script? (running ubuntu btw)

**Comment by mpscholten** [comment]
> Is there a proper way of getting the nix store directory after the command, or just some text processing by bash?

Yes, nix will automatically create a symlink at `./result` which points to the new release. So `readlink -f ./result` should help here :)

> What is the reason we need to be on ../lib?

IHP needs to find the https://github.com/digitallyinduced/ihp/tree/master/lib/IHP/static directory to serve built-in JS and CSS. If IHP is started from the `lib` directory, it is able to find that directory based on this function: https://github.com/digitallyinduced/ihp/blob/master/IHP/LibDir.hs#L14

**Comment by mpscholten** [comment]
Closing as this is not relevant for recent IHP releases anymore
